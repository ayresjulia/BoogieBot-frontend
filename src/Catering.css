.Catering {
	display: grid;
	grid-template-columns: auto auto auto auto;
	text-align: center;
	justify-content: space-evenly;
	width: auto;
}

.Catering-card {
	width: 300px;
	text-align: center;
}

.Catering-form {
	text-align: center;
	margin-top: 20px;
	margin-left: auto;
	margin-right: auto;
	width: 600px;
}

.btn {
	width: 100px;
}
