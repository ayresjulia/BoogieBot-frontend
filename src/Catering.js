import React, { useState, useEffect } from "react";
import axios from "axios";
import { Redirect } from "react-router-dom";
import Alert from "./helpers/Alert";
import SearchForm from "./forms/SearchForm";
import { Card, CardTitle, CardSubtitle, Button, CardImg, CardBody } from "reactstrap";
import "./Catering.css";
import { CLIENT_KEY_DOCUMENU } from "./secret";
import dict from "./helpers/dictionary";
import { Form, FormGroup, Label, Input, Row, Col } from "reactstrap";

const API_BASE_URL = "https://api.documenu.com/v2/restaurants/search/fields?";

const Catering = ({ events, currentUser, saveToMoodboard }) => {
	const [ restaurants, setRestaurants ] = useState([]);
	const [ checkedId, setCheckedId ] = useState(null); // get id of which event is checked
	const [ restId, setRestId ] = useState(null); // get id of which restaurant is clicked
	const [ formErrors, setFormErrors ] = useState([]);

	useEffect(() => {
		getRestaurants();
	}, []);

	async function getRestaurants (zip) {
		let getRestaurant = await axios.get(
			`${API_BASE_URL}zip_code=${zip}&key=${CLIENT_KEY_DOCUMENU}`
		);

		let data = getRestaurant.data.data;
		setRestaurants(data);
	}

	// FROM EVENT CHECKBOX FORM
	let filteredEvents = currentUser.isAdmin
		? events
		: events.filter((e) => Object.values(e).includes(currentUser.username));

	const getEventId = (e) => {
		if (e.target.checked) {
			setCheckedId(e.target.value); // got the id of checked event
		}
	};

	const getRestaurantID = (e) => {
		setRestId(e.target.value); // restaurant id
	};

	// SUBMIT FORM
	async function handleSubmit (e) {
		e.preventDefault();

		let result = await saveToMoodboard({
			event_id: parseInt(checkedId),
			inspiration_url: null,
			restaurant_key: restId
		});

		if (result.success) {
			setFormErrors(result.errors);
		}
	}

	if (!events) return <Redirect to="/" />;
	if (!currentUser.username) return <Redirect to="/" />;

	return (
		<div>
			<div className="Catering-form">
				<div className="Catering-new">
					<span className="Catering-bold">{dict.cateringLookup}</span>
					<p className="Catering-p">{dict.cateringInfo}</p>
					<p className="Catering-small">{dict.cateringFootnote}</p>
				</div>
				<SearchForm searchFor={getRestaurants} />
			</div>
			<div>
				<Form className="form-inline" onSubmit={handleSubmit}>
					<div className="Catering-row">
						<Row>
							{filteredEvents &&
								filteredEvents.map((event) => (
									<Col m={4}>
										<FormGroup check>
											<Label check>
												<Input
													type="checkbox"
													value={event.id}
													onChange={getEventId}
												/>{" "}
												{event.title}
											</Label>
										</FormGroup>
									</Col>
								))}
						</Row>
					</div>
					<div className="Catering-cards">
						{restaurants.map((item) => (
							<Card
								key={item.restaurant_id}
								id={item.restaurant_id}
								className="Catering-card">
								<CardImg
									top
									className="Catering-card-img"
									width="100%"
									src="https://images.unsplash.com/photo-1557962011-4b36a6f5d921?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1350&q=80"
									alt="restaurant image"
								/>
								<CardBody>
									<CardTitle tag="h5">{item.restaurant_name}</CardTitle>
									{item.cuisines.length > 1 && (
										<CardSubtitle tag="h6" className="mb2 text-muted">
											Cuisine: {item.cuisines[0]}
										</CardSubtitle>
									)}
									{item.address && (
										<p>
											<a
												a
												href={`http://maps.google.com/?q=${item.address[
													"formatted"
												]}`}
												target="_blank"
												rel="noreferrer"
												tag="h6"
												className="mb2 text-muted">
												{item.address["formatted"]}
											</a>
										</p>
									)}
									{item.restaurant_phone && (
										<p>
											<a
												href={`tel:${item.restaurant_phone}`}
												tag="h6"
												className="mb2 text-muted">
												{item.restaurant_phone}
											</a>
										</p>
									)}
									{item.restaurant_website && (
										<p>
											<a
												href={item.restaurant_website}
												target="_blank"
												rel="noreferrer"
												tag="h6"
												className="mb2 text-muted">
												website
											</a>
										</p>
									)}
								</CardBody>
								{formErrors.length ? (
									<Alert type="danger" messages={formErrors} />
								) : null}

								<Button
									type="submit"
									id={item.restaurant_id}
									value={item.restaurant_id}
									className="Catering-btn btn-secondary"
									onClick={getRestaurantID}>
									Save!
								</Button>
							</Card>
						))}
					</div>
				</Form>
			</div>
		</div>
	);
};

export default Catering;
