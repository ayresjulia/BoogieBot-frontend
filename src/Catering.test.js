import React from "react";
import { render } from "@testing-library/react";
import Catering from "./Catering";
import { act } from "react-dom/test-utils";
import { MemoryRouter } from "react-router";

const currentUser = { username: "testuser", isAdmin: false };

const testEvents = [
	{
		id: 1,
		title: "Anniversary",
		description: "Anniversary",
		eventDate: "2021-05-19",
		eventTime: "13:00",
		city: "Minsk",
		state: "CA",
		country: "Belarus",
		imgUrl:
			"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
		hostUsername: "testuser"
	},
	{
		id: 2,
		title: "Wedding",
		description: "Wedding",
		eventDate: "2021-05-19",
		eventTime: "13:00",
		city: "Minsk",
		state: "CA",
		country: "Belarus",
		imgUrl:
			"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
		hostUsername: "testuser"
	}
];

it("renders without crashing", async () => {
	act(() => {
		render(
			<MemoryRouter>
				<Catering events={testEvents} currentUser={currentUser} />
			</MemoryRouter>
		);
	});
});

it("shows text on the page", async () => {
	act(() => {
		const { getByText } = render(
			<MemoryRouter>
				<Catering events={testEvents} currentUser={currentUser} />
			</MemoryRouter>
		);
		expect(getByText("lookup")).toBeInTheDocument();
	});
});
