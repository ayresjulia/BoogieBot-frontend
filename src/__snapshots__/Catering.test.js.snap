// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Catering-form"
    >
      <div
        class="Catering-new"
      >
        <span
          class="Catering-bold"
        >
          lookup
        </span>
        <p
          class="Catering-p"
        >
          restaurants nearby with your zip code for catering options.
        </p>
        <p
          class="Catering-small"
        >
          check the event you want to save the restaurant to, it will appear in your event details
        </p>
      </div>
      <div
        class="SearchForm"
      >
        <form
          class="form-inline"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="search"
            />
            <input
              class="form-control flex-grow-1 form-control"
              name="search"
              placeholder="Enter search term.."
              type="text"
              value=""
            />
            <button
              class="Searchform-btn btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
    <div>
      <form
        class="form-inline"
      >
        <div
          class="Catering-row"
        >
          <div
            class="row"
          />
        </div>
        <div
          class="Catering-cards container"
        >
          <div
            class="row row-cols-1 row-cols-sm-2 row-cols-md-4"
          />
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
